src/exchange/exchange.py

```diff
@@ -111,6 +111,12 @@ def reply(self, max_tool_use: int = 128) -> Message:
             for tool_use in response.tool_use:
                 tool_result = self.call_function(tool_use)
                 content.append(tool_result)
+                if tool_result.is_error:
```

Comment:
ps in this case .is_error is true because in our validation, we catch an exception and then set this field. I don't know why we don't propagate instead, as if we did, this bug wouldn't have been possible.

If someone has an idea, I'll go back and make a comment. If they don't maybe we can in a later PR change to not swallow the error and let it break the request instead of tunneling it through a result field backwards.

Reply:
specifically here, if there was a validation error, we swallow the exception which before had us retry.

https://github.com/square/exchange/blob/main/src/exchange/exchange.py#L158-L161

I could think of some reasons to try this way, but if we did, we'd need to make sure we don't add the error into the LLM context if we did.